src/main/java/com/bmw/hmm_lib/MostLikelySequence.java

```diff
@@ -54,26 +54,38 @@
      * Since there are no previous states for t=1, backPointerSequence starts with t=2.
      */
     public final List<Map<S, S>> backPointerSequence;
+    private final int brokenTimeStep;
 
-    public MostLikelySequence(List<S> mostLikelySequence, boolean isBroken,
+    public MostLikelySequence(List<S> mostLikelySequence, boolean isBroken, int brokenTimeStep,
```

Comment:
It's probably even better to pass null as brokenTimeStep (and use Integer instead of int) if the HMM is not broken.